### PR TITLE
[feature] CDK stack setup of External Slurmdbd

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Add necessary resources to CDK stack, eg: NLB, SG, ASG, IAM role
* Add user_data script and dna.json to LT

### Tests
* Manually tests not finished yet

### References
* cookbook PR: https://github.com/aws/aws-parallelcluster-cookbook/pull/2574

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/constants.py [hunk 1]
```diff
@@ -325,3 +325,5 @@ class Operation(Enum):
 IAM_ROLE_REGEX = "^arn:.*:role/"
 IAM_INSTANCE_PROFILE_REGEX = "^arn:.*:instance-profile/"
 IAM_POLICY_REGEX = "^arn:.*:policy/"
+
+EXTERNAL_SLURMDBD_ASG_SIZE = "1"
```
Comment (jdeamicis) [inline cli/src/pcluster/constants.py]: I don't think this constant should be part of the PC codebase.

File: cloudformation/external-slurmdbd/.gitignore [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
+
+# CDK Context & Staging files
+.cdk.staging/
+cdk.out/
```
Comment (jdeamicis) [inline cloudformation/external-slurmdbd/.gitignore]: Please add an end of line char.

File: cloudformation/external-slurmdbd/source.bat [hunk 3]
```diff
@@ -0,0 +1,13 @@
+@echo off
```
Comment (jdeamicis) [inline cloudformation/external-slurmdbd/source.bat]: Please remove this file from the PR.

File: cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        )
+        self.munge_key_secret_arn = CfnParameter(
+            self, "MungeKeySecretArn", type="String", description="Secret ARN for Munge key."
+        )
+
+        # use cfn-init and cfn-hup configure instance
+        self._cfn_init_config = self._add_cfn_init_config()
+
+        # create management security group with SSH access from anywhere (TEMPORARY!)
+        self._ssh_server_sg, self._ssh_client_sg = self._add_management_security_groups()
```
Comment (jdeamicis) [inline cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py:64]: Please add another pair of security groups for the slurm accounting traffic across between cluster head node and external slurmdbd instance via port 6819.

**Review by jdeamicis — COMMENTED** [review]
Left a few comments around.

File: cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        dna_json_content = {
+            "dbms_uri": self.dbms_uri_param.value_as_string,
+            "dbms_username": self.dbms_username_param.value_as_string,
+            "dbms_database_name": self.dbms_database_name_para.value_as_string,
+            "dbms_password_secret_arn": self.dbms_password_secret_arn_param.value_as_string,
+            "munge_key_secret_arn": self.munge_key_secret_arn_param.value_as_string,
+            "region": self.stack.region,
+            "stack_name": self.stack.stack_name,
+            "nlb_dns_name": self._external_slurmdbd_nlb.load_balancer_dns_name,
+            "if_external_slurmdbd": True,
```
Comment (jdeamicis) [inline cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py]: Rename this parameter to `is_external_slurmdbd`.

File: cli/src/pcluster/constants.py (hunk 1 above)
Comment (hehe7318) [reply cli/src/pcluster/constants.py]: Have moved!

File: cloudformation/external-slurmdbd/.gitignore (hunk 2 above)
Comment (hehe7318) [reply cloudformation/external-slurmdbd/.gitignore]: Done!

File: cloudformation/external-slurmdbd/source.bat (hunk 3 above)
Comment (hehe7318) [reply cloudformation/external-slurmdbd/source.bat]: Done!

File: cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py (hunk 5 above)
Comment (hehe7318) [reply cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py]: Done!

File: cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py (hunk 4 above)
Comment (hehe7318) [reply cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py:64]: Done!

**Review by jdeamicis — APPROVED** [review]

**Pull request merged**